src/Model/Publication/Config/PublicationUsageGroupRepository.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace srag\Plugins\Opencast\Model\Publication\Config;
+
+use srag\Plugins\Opencast\LegacyHelpers\TranslatorTrait;
+
+/**
+ * Class PublicationUsageGroupRepository
+ *
+ * @author Farbod Zamani Boroujeni <zamani@elan-ev.de>
+ */
+class PublicationUsageGroupRepository
+{
+    use TranslatorTrait;
+    /**
+     * Returns the display name of the group but by looking for a record also in localization.
+     * @param int $group_id
+     * @return string
+     */
+    public static function getDisplayName(int $group_id): string
+    {
+        $display_name = '';
+        $group = PublicationUsageGroup::find($group_id);
+        if (!empty($group) && !empty($group->getDisplayName())) {
+            $display_name = $group->getDisplayName();
+            $display_name = self::getLocalizedDisplayName($display_name);
+        }
+        return trim($display_name);
+    }
+
+    /**
+     * @param string $group_display_name
+     * @return string
+     */
+    public static function getLocalizedDisplayName($group_display_name): string
+    {
+        $localized_display_name = self::translate(strtolower($group_display_name), PublicationUsageGroup::DISPLAY_NAME_LANG_MODULE);
```

Comment:
`translate` is non static

Reply:
relying on 'MISSING' could be a problem in the future. The `TranslatorTrait` is marked as deprecated, we should use `$this->plugin->txt()` instead. wich leads to the question whether a data repository already should translate or not. the txt() method won't return a MISSING string in the future.

Reply:
PS: I am afraid that TranslatorTrait comes from your work when removing DICTrait!

Reply:
Having me to cope with the latest work from ILIAS 8/9 here to prevent future work is a bit unnecessary IMHO

Reply:
Check the 'MISSING' won't break anything if it is not there, and it is somehow a complementary feature for now!

Reply:
`self::translate`must be changed to `$this->translate` anyway.

yes the Trait comes from the refactoring for locations where the plugin used the much larger translate method of the DICTrait. changing those locations will be a huge work as well. but we should not introduce more locations where we use this Trait (therefore it's declared deprecated: https://github.com/opencast-ilias/OpenCast/blob/main/src/LegacyHelpers/TranslatorTrait.php#L29 )

Reply:
I fixed the static problem, and will push that in a batch later when eveything is resolved